addons/stock/models/product.py

```diff
@@ -86,10 +87,10 @@ def _compute_quantities(self):
 
     def _product_available(self, field_names=None, arg=False):
         """ Compatibility method """
-        return self._compute_quantities_dict(self._context.get('lot_id'), self._context.get('owner_id'), self._context.get('package_id'), self._context.get('from_date'), self._context.get('to_date'))
+        return self._compute_quantities_dict(self._context.get('lot_id'), self._context.get('owner_id'), self._context.get('package_id'), self._context.get('from_date'), self._context.get('to_date'), self._context.get('quantity_available_locations_domain'))
 
-    def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False, to_date=False):
-        domain_quant_loc, domain_move_in_loc, domain_move_out_loc = self._get_domain_locations()
+    def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False, to_date=False, quantity_available_locations_domain=False):
+        domain_quant_loc, domain_move_in_loc, domain_move_out_loc = self._get_domain_locations(usage=quantity_available_locations_domain)
```

Comment:
finally, remove the kwargs
ue if self.env.conext.get(quantity_available_locations_domain) only when needed